### PR TITLE
Fix broken campaign export by ensuring getVariantSettings always returns an array

### LINKED ISSUE #15570: Campaign export is silently failing
### Mautic Series

7.0.x series

### Mautic installed version

7.0

### Way of installing

I cloned the Mautic repo (not recommended or supported)

### PHP version

8.4

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When trying to export a campaign, it silently fails - there is no error warning displayed, but the export doesn't work.

It's a bit worrying that this wasn't caught by a test on the campaign export functionality, so we should also check that is working as expected.

The campaign has an asset which has a language variant.

The campaign export is here: 

[MauticCampaignExport-2025-10-19 08_42_34.zip](https://github.com/user-attachments/files/22989464/MauticCampaignExport-2025-10-19.08_42_34.zip) however I came across #15572 when trying to import so that might not work!

### How can we reproduce this issue?

Step 1: Create a campaign with an asset which has a translated version
Step 2: Try to export the campaign

### Relevant log output

```shell
[2025-10-19T08:29:03.583976+00:00] mautic.CRITICAL: Uncaught PHP Exception TypeError: "Mautic\EmailBundle\Entity\Email::getVariantSettings(): Return value must be of type array, null returned" at /var/www/html/app/bundles/CoreBundle/Entity/VariantEntityTrait.php line 146 {"exception":"[object] (TypeError(code: 0): Mautic\\EmailBundle\\Entity\\Email::getVariantSettings(): Return value must be of type array, null returned at /var/www/html/app/bundles/CoreBundle/Entity/VariantEntityTrait.php:146)
[stacktrace]
#0 /var/www/html/app/bundles/EmailBundle/EventListener/EmailImportExportSubscriber.php(83): Mautic\\EmailBundle\\Entity\\Email->getVariantSettings()
#1 /var/www/html/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(115): Mautic\\EmailBundle\\EventListener\\EmailImportExportSubscriber->onEmailExport()
#2 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(206): Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener->__invoke()
#3 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(56): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()
#4 /var/www/html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(126): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch()
#5 /var/www/html/app/bundles/CampaignBundle/EventListener/CampaignEventImportExportSubscriber.php(132): Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher->dispatch()
#6 /var/www/html/app/bundles/CampaignBundle/EventListener/CampaignEventImportExportSubscriber.php(72): Mautic\\CampaignBundle\\EventListener\\CampaignEventImportExportSubscriber->handleChannelExport()
#7 /var/www/html/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(115): Mautic\\CampaignBundle\\EventListener\\CampaignEventImportExportSubscriber->onExport()
#8 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(206): Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener->__invoke()
#9 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(56): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()
#10 /var/www/html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(126): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch()
#11 /var/www/html/app/bundles/CampaignBundle/EventListener/CampaignImportExportSubscriber.php(76): Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher->dispatch()
#12 /var/www/html/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(115): Mautic\\CampaignBundle\\EventListener\\CampaignImportExportSubscriber->onCampaignExport()
#13 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(206): Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener->__invoke()
#14 /var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php(56): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()
#15 /var/www/html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(126): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch()
#16 /var/www/html/app/bundles/CampaignBundle/Controller/CampaignController.php(191): Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher->dispatch()
#17 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(183): Mautic\\CampaignBundle\\Controller\\CampaignController->exportAction()
#18 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(76): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()
#19 /var/www/html/vendor/symfony/framework-bundle/Controller/AbstractController.php(120): Symfony\\Component\\HttpKernel\\HttpKernel->handle()
#20 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(434): Symfony\\Bundle\\FrameworkBundle\\Controller\\AbstractController->forward()
#21 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(183): Mautic\\CoreBundle\\Controller\\CommonController->executeAction()
#22 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(76): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()
#23 /var/www/html/vendor/symfony/http-kernel/Kernel.php(182): Symfony\\Component\\HttpKernel\\HttpKernel->handle()
#24 /var/www/html/app/AppKernel.php(101): Symfony\\Component\\HttpKernel\\Kernel->handle()
#25 /var/www/html/app/middlewares/CORSMiddleware.php(76): AppKernel->handle()
#26 /var/www/html/app/middlewares/HSTSMiddleware.php(39): Mautic\\Middleware\\CORSMiddleware->handle()
#27 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(28): Mautic\\Middleware\\HSTSMiddleware->handle()
#28 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(52): Mautic\\Middleware\\CatchExceptionMiddleware->handle()
#29 /var/www/html/app/middlewares/VersionCheckMiddleware.php(58): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle()
#30 /var/www/html/app/middlewares/TrustMiddleware.php(42): Mautic\\Middleware\\VersionCheckMiddleware->handle()
#31 /var/www/html/app/middlewares/StackedHttpKernel.php(55): Mautic\\Middleware\\TrustMiddleware->handle()
#32 /var/www/html/index.php(19): Mautic\\Middleware\\StackedHttpKernel->handle()
#33 {main}
"} {"hostname":"mautic7demo-web","pid":2365}
```

### Code of Conduct

- [x] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

### PR DESCRIPTION
**Pull request by RCheesley**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✔️ 
| New feature/enhancement? (use the a.x branch)      | ❌
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ❓I will ask Copilot for help with this.
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #15570

## Description

When exporting a campaign, the export fails silently and nothing happens. As documented in #15570 there's an error thrown in the logs which suggests a problem with getVariantSettings.

This PR is entirely GitHub Copilot so please review it for accuracy - it solved my problem and I was able to export the campaign after I made this change. 

I will also ask Copilot to write an automated test unless someone would be up for helping out with that.

❓Should we throw an error in the UI in the future for this kind of error? It was a bit odd that there was nothing presented to the user.

❓Why was it not caught in automated tests for campaign export?

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Try to export a campaign with the 7.x branch - it doesn't work.
3. Apply this PR and check you can export a campaign.

### CODE REVIEW

File: app/bundles/CoreBundle/Entity/VariantEntityTrait.php [hunk 1]
```diff
@@ -141,7 +141,7 @@ public function setVariantSettings(array $variantSettings): static
      */
     public function getVariantSettings(): array
     {
-        return $this->variantSettings;
```
Comment (andersonjeccel) [inline app/bundles/CoreBundle/Entity/VariantEntityTrait.php:144]: you might want to add a regression test where `variant_settings` is null so the campaign export path stays covered. that is the case this guard is fixing.
